azavea/raster-foundry
# Rework airflow geotiff import to poll for uploads

## Overview

This PR makes the `find_geotiff_scenes` airflow DAG use `/api/uploads/` to find scenes to import.

### Checklist


## Testing Instructions

- create an upload from the api console (`:paste` this):
```scala
/** Prepare a bunch of values */
val user = Await.result(Users.getUserById("default"), 3 seconds).get
val orgId = Await.result(database.db.run(Organizations.take(1).result), 3 seconds).head.id
val datasourceId = Await.result(database.db.run(Datasources.filter(_.visibility === (Visibility.Public:Rep[Visibility])).take(1).result), 3 seconds).head.id
val uploadStatus = UploadStatus.fromString("uploaded")
val fileType = FileType.fromString("geotiff")
val uploadType = UploadType.fromString("s3")
val files = List.empty[String]
val metadata = ().asJson
/** Actually create the Upload */
val uploadCreate = Upload.Create(orgId, uploadStatus, fileType, uploadType, files, datasourceId, metadata)
Await.result(database.db.run(Uploads.insertUpload(uploadCreate, user)), 3 seconds)
/** :tada: */
```
- fire up your server with airflow `./scripts/server --with-airflow`
- in another terminal, `docker-compose -f docker-compose.airflow.yml up airflow-worker`. This is so you can see the airflow-worker logs without all the cruft around them
- once everything is up, `docker-compose -f docker-compose.airflow.yml run airflow-scheduler bash`
- Run the task to find geotiff scenes: `airflow run find_geotiff_scenes find_new_geotiff_scenes 2017-03-29` (date doesn't matter)
- Navigate to Browse / Task Instances in your airflow UI
- The `find_geotiff_scenes` dag should run, and you should get a failed run for `import_geotiff_scenes`

Connects #1301
